plugins/volume/org.xen.xapi.storage.zfs-ng/sr.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python
+
+import importlib
+import os
+import os.path
+import sys
+import urlparse
+
+from xapi.storage import log
+from xapi.storage.libs import util
+from xapi.storage.common import call
+from xapi.storage.libs.libcow.volume import COWVolume
+import xapi.storage.api.v5.volume
+
+import zfsutils
+
+
+@util.decorate_all_routines(util.log_exceptions_in_function)
+class Implementation(xapi.storage.api.v5.volume.SR_skeleton):
+
+    def create(self, dbg, sr_uuid, configuration, name, description):
+        log.debug('{}: SR.create: config={}, sr_uuid={}'.format(
+            dbg, configuration, sr_uuid))
+
+        # 2 ways to create a SR:
+        # - from an existing zpool (manually created for complex configs)
+        # - from a "devices" config string (comma-separated list of devices)
+
+        if 'zpool' in configuration:
+            if 'devices' not in configuration:
```

Comment:
it is if 'devices' in configuration I think

Reply:
yes, that was corrected in a later fixup - everything has been squashed now